parsers/cxx/cxx_parser_tokenizer.c

```diff
@@ -1079,29 +1136,58 @@ bool cxxParserParseNextToken(void)
 				}
 			}
 		} else {
-			bool bIgnoreParens = false;
-			const char * szReplacement = NULL;
-			if(isIgnoreToken(
-					vStringValue(t->pszWord),
-					&bIgnoreParens,
-					&szReplacement
-				))
+			// We can enter here also from a jump to check_keyword after finding
+			// and ignored token and a replacement. 
+
+			if(!szReplacement)
 			{
-				CXX_DEBUG_PRINT("Ignore token %s",vStringValue(t->pszWord));
-				// FIXME: Handle ignore parens!
-				if(szReplacement && *szReplacement)
+				bool bIgnoreParens = false;
+	
+				if(
+					isIgnoreToken(
+							vStringValue(t->pszWord),
+							&bIgnoreParens,
+							&szReplacement
+						)
+					)
 				{
-					vStringClear(t->pszWord);
-					vStringCatS(t->pszWord,szReplacement);
-				} else {
-					// skip
-					cxxTokenChainDestroyLast(g_cxx.pTokenChain);
+					CXX_DEBUG_PRINT("Ignore token %s",vStringValue(t->pszWord));
+	
+					if(szReplacement && *szReplacement)
+					{
+						CXX_DEBUG_PRINT(
+								"The token has replacement %s: applying",
+								szReplacement
+							);
+						vStringClear(t->pszWord);
+						vStringCatS(t->pszWord,szReplacement);
+						// FIXME: Do we need to interpret the replacement?
+					} else {
+						// kill it
+						CXX_DEBUG_PRINT("Ignore token has no replacement");
+						cxxTokenChainDestroyLast(g_cxx.pTokenChain);
+					}
+	
+					if(bIgnoreParens)
+					{
+						CXX_DEBUG_PRINT("Ignored token specifies to ignore parenthesis");
+						if(!cxxParserParseNextTokenSkipIgnoredParenthesis())
+							return false;
+					}
+	
+					if(szReplacement && *szReplacement)
+					{
+						// Already have a token to return.
+						// Check again for keywords.
+						goto check_keyword;
+					}
+	
 					return cxxParserParseNextToken();
 				}
 			}
 		}
 
-		t->bFollowedBySpace = t->bFollowedBySpace | isspace(g_cxx.iChar);
+		t->bFollowedBySpace = isspace(g_cxx.iChar);
```

Comment:
If I'm not wrong that t->bFollowedBySpace should always read to false on that line.